subprojects/docs/src/docs/userguide/javaLibraryPlugin.xml

```diff
@@ -0,0 +1,276 @@
+            <listitem>dependencies do not leak into the compile classpath of consumers anymore, so you will never accidently depend on a transitive dependency</listitem>
+            <listitem>faster compilation thanks to reduced classpath size</listitem>
+            <listitem>less recompilations when implementation dependencies change: consumers would not need to be recompiled</listitem>
+            <listitem>cleaner publishing: when used in conjunction with the new <literal>maven-publish</literal> plugin, Java libraries produce POM files that distinguish exactly between what
+                is required to compile against the library and what is required to use the library at runtime (in other words, don't mix what is needed to compile
+                the library itself and what is needed to compile against the library).
+            </listitem>
+        </itemizedlist>
+        <note>The <literal>compile</literal> configuration still exists but should not be used as it will not offer the guarantees that the
+            <literal>api</literal>
+            and <literal>implementation</literal> configurations provide.
+        </note>
+    </section>
+
+    <section id="sec:java_library_recognizing_dependencies">
+        <title>Recognizing API and implementation dependencies</title>
+        <para>As soon as we want to separate API and implementation dependencies, we require some guidelines as to make the difference between the two.
+            Basically, an API dependency is a type that is exposed in the library binary interface, often referred to ABI (Application Binary Interface).
+            This includes, but is not limited to:
+        </para>
+        <itemizedlist>
+            <listitem>types used in super classes or interfaces</listitem>
+            <listitem>types used in public method parameters, including generic parameter types (where <firstterm>public</firstterm> is something that is visible to compilers.
+                I.e. , <firstterm>public</firstterm>, <firstterm>protected</firstterm> and <firstterm>package private</firstterm> members in the Java world)
+            </listitem>
+            <listitem>types used in public fields</listitem>
+            <listitem>public annotation types</listitem>
+        </itemizedlist>
+        <para>In opposition, any type that is used in the following list is irrelevant to the ABI, and therefore should be declared as <literal>implementation</literal> dependency:
+        </para>
+        <itemizedlist>
+            <listitem>types exclusively used in method bodies</listitem>
+            <listitem>types exclusively used in private members</listitem>
+            <listitem>types exclusively found in internal classes (future versions of Gradle will let you declare which packages belong to the public API)</listitem>
+        </itemizedlist>
+        <para>In the following sample, we can make the difference between an API dependency and an implementation dependency:</para>
```

Comment:
I'm happy to consider alternate examples, I came with no better. I wanted a _meaningful_, _understandable_, sample, that people could use. It's surprisingly hard to find an example that uses both API and implementation dependencies, which is also compact and understandable. As for the imports, I disagree: the idea of this chapter is to tell "you should separate API and impl", then "here's how you can make the difference".